src/intervals/rounding.jl

```diff
@@ -35,7 +47,26 @@ macro round(T, expr1, expr2)
             Interval(prevfloat($expr1), nextfloat($expr2))
 
         else # mode == :narrow
-            Interval(@with_rounding($T, $expr1, RoundDown), @with_rounding($T, $expr2, RoundUp))
+
+            # The idea here is, if RoundNearest is not equal to
+            # RoundDown or RoundUp, then the (Float64) operation does
+            # not round too sharply
+            lnea = @with_rounding($T, $expr1, RoundNearest)
+            ldow = @with_rounding($T, $expr1, RoundDown)
+            lup = @with_rounding($T, $expr1, RoundUp)
+            lo = ifelse(lnea != ldow && lnea != lup, lnea, ldow)
```

Comment:
That's in the comment above :-) The idea is that the correct rounding should leave on of the roundings fixed to nearest, unless you really get the zero-distance for nearest rounding. It yields quite consistent things with respect to the IRF1788 tests.